src/OpenTelemetry/Internal/SelfDiagnosticsConfigRefresher.cs

```diff
@@ -44,12 +45,24 @@ internal class SelfDiagnosticsConfigRefresher : IDisposable
 
     public SelfDiagnosticsConfigRefresher()
     {
-        this.configParser = new SelfDiagnosticsConfigParser();
-        this.UpdateMemoryMappedFileFromConfiguration();
+        if (this.IsSelfDiagnosticsEnVarOn)
+        {
+            this.envarParser = new SelfDiagnosticsEnVarParser();
+            this.logEventLevel = this.envarParser.GetLogLevel();
+            this.eventListener = new SelfDiagnosticsEventListener(this.logEventLevel, this);
+        }
+        else
+        {
+            this.configParser = new SelfDiagnosticsConfigParser();
+            this.UpdateMemoryMappedFileFromConfiguration();
+        }
+
         this.cancellationTokenSource = new CancellationTokenSource();
         this.worker = Task.Run(() => this.Worker(this.cancellationTokenSource.Token), this.cancellationTokenSource.Token);
```

Comment:
Is it necessary to periodically check the config file if env var is set? What about if both env var and config file are present?